addons/survey/models/survey_user.py

```diff
@@ -276,6 +281,29 @@ def _get_line_comment_values(self, question, comment):
             'value_char_box': comment,
         }
 
+    def _get_answer_score_values(self, question, answer, answer_type):
+        # default and non-scored questions
+        vals = {'answer_score': 0, 'answer_is_correct': False}
+        # record selected suggested choice assigned score (can be: pos, neg, or 0)
+        if answer_type in ['suggestion']:
+            if answer:
+                vals.update({'answer_score': self.env['survey.question.answer'].browse(int(answer)).answer_score,
+                             'answer_is_correct': self.env['survey.question.answer'].browse(int(answer)).is_correct})
```

Comment:
Avoid browsing the same thing multiple times if you can help it.
Think of it as a performance drawback.

Save the question_answer in a variable instead.

Reply:
Understood and fixed it.